This project was borne out of a stroll with a friend through Los Angeles's Koreatown after a late-night Korean barbecue feast. in the space of just five blocks, we passed nine different Korean barbecue restaurants, three ice-cream shops, three tea houses, and so many bars I lost count. I wondered aloud to her how two K-BBQ restaurants could be located in the same plaza and not drain each other's business, which led to two burning questions I had to answer:
* How many Korean BBQ restaurants are in Koreatown, anyway?
* How could two Korean BBQ restaurants exist in the same plaza or on the same block without siphoning business away from each other? 

Koreatown is one of Los Angeles's most densely-populated and ethnically-diverse neighborhoods. In the 4 sq miles bounded by <>, 

In [91]:
import pandas as pd
import numpy as np

In [2]:
la_biz = pd.read_csv('datasets/Listing_of_Active_Businesses_LA.csv')

In [3]:
la_biz[0:30]

LOCATION ACCOUNT #                                      BUSINESS NAME  \
0   0000000115-0001-3                                   VINCENZO LABELLA   
1   0000000150-0001-5      A A OFICINA CENTRAL HISPANA DE LOS ANGELES /C   
2   0000000156-0001-2  SPRINGBOARD NON-PROFIT CONSUMER CREDIT MANAGEMENT   
3   0000000247-0001-1      A A OFICINA CENTRAL HISPANA DE LOS ANGELES /C   
4   0000000267-0001-9      A A OFICINA CENTRAL HISPANA DE LOS ANGELES /C   
5   0000000282-0001-3      A A OFICINA CENTRAL HISPANA DE LOS ANGELES /C   
6   0000000327-0001-2      A A OFICINA CENTRAL HISPANA DE LOS ANGELES /C   
7   0000000328-0001-7      A A OFICINA CENTRAL HISPANA DE LOS ANGELES /C   
8   0000000350-0001-1      A A OFICINA CENTRAL HISPANA DE LOS ANGELES /C   
9   0002990781-0001-5                                   CORALIE WHITCOMB   
10  0000000547-0001-6                        NEW CITY LIFE FOUNDATION /C   
11  0000000590-0001-2  LOS ANGELES URBAN LEAGUE HEAD START STATE PRES...   
12  0000000699-0001-1              THE KABBALAH CENTRE INTERNATIONAL INC   
13  0000000726-0001-1                                      JOYCE WILLING   
14  0003008665-0001-4                               JOSE DEJESUS SOLTERO   
15  0000001168-0001-2                         THE SKID ROW HOUSING TRUST   
16  0000001192-0001-6  THE PUBLIC HEALTH FOUNDATION OF LOS ANGELES CO...   
17  0000001271-0001-2                          ST ANTHONY MEDICAL CENTER   
18  0000001333-0001-5                            YIDDISHKAYT LOS ANGELES   
19  0000001345-0001-8                                  S RICHARD FRIESEN   
20  0000001360-0001-2  SAN FERNANDO VALLEY COMMUNITY MENTAL HEALTH CE...   
21  0000001364-0001-1                          WET PAINT PRODUCTIONS LLC   
22  0000001428-0001-2                                     JESSICA STRAUS   
23  0000001540-0001-1                      CALIFORNIA CHILDREN'S ACADEMY   
24  0000001543-0001-5                      CALIFORNIA CHILDREN'S ACADEMY   
25  0000002144-0001-4                              LAS PALMAS FOUNDATION   
26  0000001576-0001-0                                    ODILIA TRUJILLO   
27  0000001581-0001-1                                   DARIUSH DIVSALAR   
28  0000001598-0001-6                         ATLANTIC RECOVERY SERVICES   
29  0000001632-0001-5                         ATLANTIC RECOVERY SERVICES   

                                   DBA NAME                   STREET ADDRESS  \
0                                       NaN            521 SWARTHMORE AVENUE   
1                                       NaN             2015 W TEMPLE STREET   
2            MONEY MANAGEMENT INTERNATIONAL        1605 W OLYMPIC BLVD #9023   
3                                       NaN                 3806 W PICO BLVD   
4                                       NaN                 3115 VENICE BLVD   
5                                       NaN              2601 W PICO BLVD #2   
6                                       NaN                4151 BEVERLY BLVD   
7                                       NaN           3049 W 8TH STREET #763   
8                                       NaN                 1150 VENICE BLVD   
9                                       NaN             2669   MEDLOW AVENUE   
10                                      NaN         1201 W MANCHESTER AVENUE   
11                                      NaN            10611 S HOOVER STREET   
12                                      NaN         1220 S SHENANDOAH STREET   
13                                      NaN              10510 ODESSA AVENUE   
14  SERVICES SOLUTIOINS PAINTING & CLEANING      2716  1/2   LEAFDALE AVENUE   
15                                      NaN                 114 E 5TH STREET   
16     PUBLIC HEALTH FOUNDATION WIC PROGRAM          1836 W IMPERIAL HIGHWAY   
17                                      NaN           837 W IMPERIAL HIGHWAY   
18                                      NaN          3780 WILSHIRE BLVD #910   
19                           BIRDBRAIN ARTS   

In [4]:
la_biz.shape

(514878, 16)

## How many businesses are in Koreatown?

This dataset contains all currently active businesses in Los Angeles, updated mid-January 2019. 

More info [here](https://data.lacity.org/A-Prosperous-City/Listing-of-Active-Businesses/6rrh-rzua); from https://data.lacity.org

Columns in this dataset:
```
'LOCATION ACCOUNT #'
'BUSINESS NAME'
'DBA NAME'
'STREET ADDRESS'
'CITY'
'ZIP CODE'
'LOCATION DESCRIPTION'
'MAILING ADDRESS'
'MAILING CITY'
'MAILING ZIP CODE'
'NAICS'
'PRIMARY NAICS DESCRIPTION'
'COUNCIL DISTRICT'
'LOCATION START DATE'
'LOCATION END DATE'
'LOCATION'
```
At first glance: 
* There are 514,878 businesses in this dataset, with 16 columns.
* It includes metropolitan-area listings (i.e.: Glendale, Wilmington(!), Pacific Palisades, Fullerton(!!)...). We just want listings inside Koreatown, within Los Angeles city limits.
* Mailing addresses are included. Columns related to mailing addresses (`'MAILING ADDRESS', 'MAILING CITY', 'MAILING ZIP CODE'`) will not be needed for this analysis.
* The `'LOCATION END DATE'` column appears to be full of `NaN`. As this is a collection of active businesses, this column, being unnecessary, will also be dropped.
* NAICS appears to be a Dewey Decimal System-like categorization scheme for businesses.
* Column names will need to be converted to Pythonic style for ease of use.
* This collection also includes single family housing constructions (NAICS code 233210).
* Geolocations accompany some of the listings (in the `'LOCATION'` column).

Additional facts:
* Many of Los Angeles's neighborhoods (mostly in the San Fernando Valley) are treated like separate cities by the USPS and residents. Woodland Hills, Hollywood, Northridge, Studio City, _et al_. all lie within Los Angeles city limits. Not relevant to this particular analysis, but useful for understanding composition of Los Angeles neighborhoods.

In [90]:
la_biz.drop(axis=1, labels=['MAILING ADDRESS',
                            'MAILING CITY',
                            'MAILING ZIP CODE',
                            'LOCATION END DATE'], inplace=True)

KeyError: "['MAILING ADDRESS' 'MAILING CITY' 'MAILING ZIP CODE' 'LOCATION END DATE'] not found in axis"

In [6]:
la_biz.columns = [col.lower().replace(' ','_') for col in la_biz.columns]

In [ ]:
la_biz.info()

In [ ]:
pd.to_datetime(la_biz.location_start_date)

In [7]:
# all listings in los angeles proper, not including the SF valley

la_biz.loc[la_biz['city'] == 'LOS ANGELES', :].shape

(254197, 12)

Out of 514,878 listed businesses, just under half are located within city limits and outside the San Fernando Valley. We can narrow this down further by ZIP code to obtain just Koreatown listings:

In [8]:
# zip+4 information is repeated in 'location_description'
# column, so not needed here

la_biz['zip_code'] = (la_biz['zip_code']
                      .str.split(n=1, pat='-', expand=True)
                      .iloc[:,0])

In [63]:
# all listings in koreatown, by zip code (90010, 90005, 90006, acc. to wikipedia)
# will need refining later

ktown_biz = la_biz.loc[((la_biz['city'] == 'LOS ANGELES')
                        & ((la_biz['zip_code'] == '90005')
                           | (la_biz['zip_code'] == '90006')
                           | (la_biz['zip_code'] == '90004')
                           | (la_biz['zip_code'] == '90010')
                           | (la_biz['zip_code'] == '90020') 
                           | (la_biz['zip_code'] == '90019'))
                       ), :]

In [64]:
ktown_biz.shape

(32309, 12)

Within these ZIP codes, Koreatown contains 32309 businesses. 

## What is the business composition of the neighborhood?
According to the NAICS description accompanying the listings, the most common business by type in Koreatown is 'lessors of real estate (including mini warehouses & self-storage units)' at 1863, or 10.20% of the total.
* A whopping 8.11% of all the businesses in Koreatown (1,482) have no NAICS classification. They have been saved to a separate dataframe for further investigation. 
* 'Full-service restaurants' (517, or 2.83% of total Koreatown listings) are beaten out by 'All other personal services' (1384, or 7.58%), legal services (673 or 3.68%), and janitorial services (483, or 3.50%)
* 'All other personal services' warrants closer investigation.

In [65]:
ktown_biz['primary_naics_description'].value_counts(dropna=False)

Lessors of real estate (including mini warehouses & self-storage units)                                                                                           3744
NaN                                                                                                                                                               2721
All other personal services                                                                                                                                       2348
Janitorial services                                                                                                                                               1237
Independent artists, writers, & performers                                                                                                                        1079
All other professional, scientific, & technical services                                                                                                           91

In [66]:
ktown_naics_comp = ktown_biz.loc[:,['naics','primary_naics_description']]

#ktown_naics_comp_unique = pd.DataFrame(ktown_biz['naics'].value_counts(normalize=True, dropna=False))
#ktown_naics_comp_unique['primary_naics_description'] = ktown_biz['primary_naics_description'].value_counts(normalize=True, dropna=False).values
#ktown_naics_comp_unique
#ktown_naics_comp.sort_values(['primary_naics_description'])


In [67]:
ktown_biz['naics'].value_counts(normalize=True, dropna=False).index[:20]

Float64Index([531100.0,      nan, 812990.0, 561720.0, 711510.0, 541990.0,
              541100.0, 722110.0, 541600.0, 512100.0, 531310.0, 621610.0,
              611000.0, 541400.0, 452000.0, 621111.0, 233210.0, 424990.0,
              453990.0, 561500.0],
             dtype='float64')

In [68]:
ktown_naics_nulls = ktown_biz.loc[(ktown_biz['naics'].isnull()), :]
ktown_naics_nulls

location_account_#                                      business_name  \
746     0000153627-0001-6                                      ALAN NAKAGAWA   
845     0000180859-0001-0  JOHN HUNTER JR, GEO H MOORE LOGGING & LUMBER, ...   
880     0000273948-0002-9                              BOUNNA & JENNY BOULOM   
1018    0002040265-0001-4                             TOMASA JIMENEZ GUILLEN   
1029    0000833326-0002-1                                     RAFAEL RAMIREZ   
1154    0002012892-0001-6                                      YANET G LOPEZ   
1291    0002029927-0001-9                                    JUAN SANTILLANA   
1377    0002033892-0001-6                                       HARMON SMITH   
1395    0002035553-0001-6                                     ELIA RODRIGUEZ   
1452    0002040788-0001-6                                   HAMID AFRASIYABI   
1502    0002044460-0001-7                                       XIANSHU SHEN   
1528    0002046436-0001-8                                         BOK S. LEE   
1570    0002052153-0001-4                       KOREAN CHRISTIAN MISSION USA   
1626    0002058434-0001-0  UNIFICATION OF KOREAN ACADEMIC RESEARCH FOUNDA...   
1657    0002062126-0001-3                                 CELIA CE HERNANDEZ   
1659    0002062219-0001-1                               JUSTO FUENTES OROZCO   
1662    0002062630-0001-1                                 HILDA ALEJO MOLINA   
1672    0002063855-0001-4                                  EMERITA RODRIGUEZ   
1696    0002068259-0001-0                                       JOHN SOBERAL   
1714    0002072688-0001-1                                      GERARDO LOPEZ   
1726    0002074124-0001-5                             GRAND BREW CORPORATION   
1769    0002080480-0001-1                                      GARY LEE HALL   
1784    0002083118-0001-9                                MIRIAN A MONTERROSA   
1791    0002084027-0001-7                                 HYUNG D. ALLEN KIM   
1793    0002084047-0001-4                                      OFELIA E. ZEA   
1847    0002087543-0001-9                                   FELICIANO ANZORA   
1866    0002089252-0001-8                                          MDA GROUP   
1891    0002091286-0001-5                               MARIA DOLORES FLORES   
2000    0002099882-0001-4                                WONDWOSSEN D DIKRAN   
2009    0002100818-0001-9                                ZACCARI STYLING INC   
...                   ...                                                ...   
510600  0003091585-0001-0                                            MIN KIM   
510682  0003092328-0001-2                                  HIK GLONICS, INC.   
510759  0003081424-0001-9                                    MNK STUDIO INC.   
510925  0003091283-0001-6                        CAFE PARISIEN LARCHMONT LLC   
511188  0002504657-0001-3                     COX DIGIOVANNI ENTERPRISES INC   
511296  0003088773-0001-8                      DODO MAMA KITCHEN CORPORATION   
511467  0003094385-0001-5                                    G & H MEDIA INC   
511559  0000389640-0005-8                                   HARRY BARSEGHIAN   
511649  0003091922-0001-7                                       JIN KOO BACK   
512021  0003091651-0001-9                                         OONA APPEL   
512317  0003092287-0001-1                                STYLE PUDDING, INC.   
512573  0002919860-0001-6                                      JENNER PASCUA   
512814  0000974126-0034-1                           DOLEX DOLLAR EXPRESS INC   
512904  0003091952-0001-8                                     MORGAN CANAVAN   
512955  0002577926-0001-9                                      JACKSON ADAMS   
512996  0003090511-0001-9                                       MOO JIN PARK   
513027  0003091249-0001-9                              KBY FEDORA PLAZA, LLC   
513035  0002969357-0001-5                                      WILLIAM START   
513113 

## How many bars are in Koreatown?

Where do bars fall in this list of categories? The `'business_name'` and `'dba_name'` columns should yield answers to this question and to the different types of restaurants within K-Town. I can also search for a known bar elsewhere in the listings.

In [14]:
la_biz.loc[la_biz['dba_name'].isin(['IDLE HOUR']), :]

location_account_# business_name   dba_name        street_address  \
333646  0000096461-0003-5      BA2, LLC  IDLE HOUR  4824 VINELAND AVENUE   

                   city zip_code      location_description     naics  \
333646  NORTH HOLLYWOOD    91601  4824 VINELAND 91601-3828  722410.0   

                    primary_naics_description  council_district  \
333646  Drinking places (alcoholic beverages)                 2   

       location_start_date             location  
333646          02/13/2015  (34.158, -118.3702)

In [15]:
la_biz.loc[la_biz['dba_name'].isin(['SASSAFRAS']), :]

location_account_# business_name   dba_name    street_address  \
111735  0002431994-0002-9       BA2 LLC  SASSAFRAS  1231 VINE STREET   

               city zip_code  location_description     naics  \
111735  LOS ANGELES    90038  1231 VINE 90038-1621  445310.0   

         primary_naics_description  council_district location_start_date  \
111735  Beer, wine & liquor stores                13          01/01/2012   

                    location  
111735  (34.0935, -118.3266)

In [16]:
la_biz.loc[(la_biz['dba_name'].isin(['THE FEDERAL BAR'])), ['naics','primary_naics_description']]

naics primary_naics_description
201376  722110.0  Full-service restaurants

In [17]:
la_biz.loc[(la_biz['dba_name'].str.contains('BBQ') == True), :]

location_account_#                 business_name  \
8905    0003010103-0001-1  GOLDEN FIELD RESTAURANTS LLC   
10963   0002427057-0001-3        SANTA MONICA BUNDY LLC   
11854   0003011656-0001-1  AMALGAMATED BBQ AND BEVERAGE   
14562   0002810274-0001-8       L A RSIAN BBQ GRILL LLC   
15091   0000814301-0001-7   KAM SHUN LAU/CHE CHEUK WONG   
15329   0002632851-0001-2                  AY-DO-NO INC   
16667   0003011242-0001-6             WEIXING GROUP INC   
17819   0002361518-0001-1         YOUNG JIN CORPORATION   
18647   0002713796-0001-1           HORSE THIEF BBQ LLC   
31793   0002051853-0001-4               SILVIOS 100 LLC   
33808   0003012303-0001-1             BILLY R TURNER JR   
40532   0002679483-0001-6          GK FOOD SERVICES INC   
45138   0002269323-0001-8          SAN PEDRO GAFFEY LLC   
46302   0002165060-0001-0       AL & LEX INVESTMENT INC   
46665   0003021202-0001-8                   JAMES T KIM   
56518   0000356078-0001-2    THAI BBQ OF VENTURA BL INC   
62074   0003021631-0001-7                    GUHARA INC   
64554   0003025243-0001-6                   ACEPLUS INC   
66899   0002683382-0001-6            APELILA AND J, LLC   
67890   0002692589-0001-3                   ALEX MILLER   
68786   0002421273-0001-5               KOGI GROUP CORP   
83107   0003020653-0001-6                JONATHAN BERRY   
84892   0003027896-0001-1                     XINDA GUO   
88287   0002730146-0001-5              MAX CITY BBQ LLC   
90053   0002569863-0001-0           JHY ENTERPRISES INC   
93334   0002436114-0001-2               RAMSHIN DANESHI   
95860   0002579482-0001-6             MANUEL E GONZALEZ   
98921   0002721014-0001-8                   KNUSAHA LLC   
105464  0000934936-0002-6               BUDNAMU USA INC   
113481  0002648250-0001-9                      CYTO INC   
...                   ...                           ...   
418613  0002979490-0001-2          CHANG FOOD GROUP INC   
421659  0002975309-0001-3                 PHILIP MARTIN   
428241  0002683382-0003-2            APELILA AND J, LLC   
432917  0002683382-0004-1            APELILA AND J, LLC   
441665  0002731220-0002-4               EILEEN MARTINEZ   
441712  0002995276-0001-0            URBAN BARBECUE LLC   
455225  0000263862-0001-6              SUVARNABHUMI INC   
458317  0003002429-0001-8      TMC RESTAURANT GROUP INC   
467636  0002914625-0001-1               BREAKERS VC LLC   
467836  0003004888-0001-2         STAUNTON AND STAUNTON   
467988  0002958119-0001-0                EUNSANG 75 INC   
470053  0000449228-0001-1         HANKOOK BBQ HOUSE INC   
473165  0002983700-0001-7                 DYL BBQ, INC.   
474494  0002995158-0001-3                  SANDRA RILEY   
474576  0002727084-0001-7                ROBERT RAMIREZ   
475109  0002907153-0001-8                 KATHY CARLSON   
476820  0002937614-0001-8                     MIDRI INC   
476981  0002900946-0001-2             GEN NORTHRIDGE LP   
481775  0002946132-0001-4             FERNANDO CARRILLO   
482262  0002639469-0001-1                   MIDRI, INC.   
488341  0002547673-0001-0                JOSE A CORCINO   
493044  0002912220-0001-1            SAMUEL E GORDON JR   
497044  0002769822-0001-1                    GYB LA LLC   
500608  0002908630-0001-5           RODQUI INCORPORATED   
506425  0002952695-0001-8                   MS SUN CORP   
507449  0002789624-0001-7            DIDIM FOOD USA INC   
510750  0003090950-0001-1         GM EVENT SERVICES LLC   
513999  0003088937-0001-8           OHB RESTAURANT, LLC   
514100  0003088937-0003-4           OHB RESTAURANT, LLC   
514760  0003088937-0002-6           OHB RESTAURANT, LLC   

                                      dba_name  \
8905                                BBQ + RICE   
10963                         ONO HAWAIIAN BBQ   
11854                                 CAFE BBQ   
14562                      L A RSIAN BBQ GRILL   
15091               GOLDEN CITY BBQ RESTAURANT   
15329                      AY-DO-NO KOREAN BBQ  

Upon further investigation (searching for bars and restaurants by name), bars are most likely to have one of three different NAICS codes, depending on whether they serve food (or no NAICS code at all):
```
722110.0	Full-service restaurants
445310.0	Beer, wine & liquor stores
722410.0	Drinking places (alcoholic beverages)	
```
I will perform an aggregation by these three categories and through `ktown_naics_nulls` as well.

In [18]:
ktown_drink_places = ktown_biz.loc[(ktown_biz['naics'] == 722410.0), :]
ktown_drink_places

location_account_#                 business_name  \
5303    0000753897-0001-1          ROSALVA M DE HOLGUIN   
12076   0002938826-0001-4              GOURMET GURU INC   
26223   0002893116-0001-0      Y & K ENTERTAINMENT CORP   
31595   0003014052-0001-5               WHITE AND BROWN   
49911   0002468971-0001-0                   ACEPLUS INC   
206940  0002397318-0002-5  S & J UNITED ENTERPRISES INC   
223907  0002687906-0001-3               TWITTERLAND INC   
269971  0000102720-0001-3           PLOENPIDH P HOUSTON   
290675  0003066599-0001-9                     SWO, INC.   
291593  0000574905-0001-6                  YONG CHA KIM   
312464  0002796240-0001-8                SOUTHLAND BEER   
351282  0000192846-0001-3             CIPRIANA M MULATO   
368331  0000294060-0001-8    KDK INVESTMENT CORPORATION   
398479  0000041953-0001-7                       BNE INC   
423242  0002887550-0001-2              FRANK N HANK LLC   
442516  0000635321-0001-3     WILLIAM/MARGARITA ALMEIDA   
458209  0002840424-0001-5         TEUM RESTAURANT, INC.   
483570  0002985640-0001-8             CHOONG CHUNG CORP   

                                              dba_name  \
5303                                     MIKE'S TAVERN   
12076   GONG CHA KOREA TOWN | GONGA CHA - LITTLE TOKYO   
26223                                              JAM   
31595                                              NaN   
49911                                            STORY   
206940                                    CLUB BENTLEY   
223907                                             NaN   
269971                                     MONTE CARLO   
290675                                       BOOLIE YA   
291593                                 CLUB TEMPTATION   
312464                                             NaN   
351282                                   APULO REST #2   
368331                                           FERIA   
398479                                       HONEY PIG   
423242                                             NaN   
442516                                      RODEO ROOM   
458209                                             NaN   
483570                                             NaN   

                            street_address         city zip_code  \
5303                      2667 W PICO BLVD  LOS ANGELES    90006   
12076   3785   WILSHIRE BLVD   SUITE #107E  LOS ANGELES    90010   
26223                    3981 W 6TH STREET  LOS ANGELES    90020   
31595   603 S MARIPOSA AVENUE   SUITE #101  LOS ANGELES    90005   
49911                    4001 W 6TH STREET  LOS ANGELES    90020   
206940                    3609 W PICO BLVD  LOS ANGELES    90019   
223907        3465 W 6TH STREET SUITE #200  LOS ANGELES    90020   
269971                   3514 W 3RD STREET  LOS ANGELES    90020   
290675  3959   WILSHIRE BLVD   SUITE #B211  LOS ANGELES    90010   
291593            739 1/2 S WESTERN AVENUE  LOS ANGELES    90005   
312464      740 S WESTERN AVENUE UNIT #112  LOS ANGELES    90005   
351282                   3958 W 6TH STREET  LOS ANGELES    90020   
368331               3470 WILSHIRE BLVD #1  LOS ANGELES    90010   
398479                   3400 W 8TH STREET  LOS ANGELES    90005   
423242                518 S WESTERN AVENUE  LOS ANGELES    90020   
442516               1306 S VERMONT AVENUE  LOS ANGELES    90006   
458209                   3526 W 8TH STREET  LOS ANGELES    90005   
483570                984 S WESTERN AVENUE  LOS ANGELES    90006   

            location_description     naics  \
5303        2667 PICO 90006-3918  722410.0   
12076   3785 WILSHIRE 90010-2899  722410.0   
26223        3981 6TH 90020-4209  722410.0   
31595         603 MARIPOSA 90005  722410.0   
49911        4001 6TH 90020-4403  722410.0   
206940      3609 PICO 90019-3426  722410.0   
223907       3465 6TH 90020-2575  722410.0   
269971       3514 3RD 90020-2002  722410.0   
290675       3959 WILSHIRE 90010  722410.0   
291593    739 WESTERN 90005-3

In [19]:
ktown_drink_places.shape

(18, 12)

In [20]:
ktown_restaurants = ktown_biz.loc[(ktown_biz['naics'] == 722110.0), :]
ktown_restaurants

location_account_#                       business_name  \
2378    0000075873-0001-4                   J & U CORPORATION   
2629    0000295584-0001-1                      ESTELA SERRANO   
4162    0002836573-0001-4               8TH AND HOOVER EB5 LP   
7406    0002738450-0001-4                SSAM RESTAURANT, INC   
8599    0000001176-0001-7       KOREAN GARDENS RESTAURANT INC   
10158   0003009401-0001-5             THE NORMANDIE GROUP LLC   
10600   0002585344-0001-0                    SEONG CHEOL SHIN   
11882   0002574693-0001-4                 BUL SAMGYE TANG INC   
13111   0000991047-0001-9               WILSHIRE KINGSLEY INC   
14793   0000558001-0001-1        CHOSUN GALBEE RESTAURANT INC   
16486   0003012148-0001-2                     GOLTONG CHICKEN   
16830   0003010987-0001-3       CLAYTON AND CASEY CORPORATION   
17242   0002603928-0001-0                         CHUN ZI KUK   
17667   0002986166-0001-9                   CHIMAC STAR, INC.   
17819   0002361518-0001-1               YOUNG JIN CORPORATION   
17867   0000071427-0001-2            OCHA CLASSIC CORPORATION   
18431   0003016013-0001-0                       OLY-CAL, INC.   
18495   0000645077-0001-1                        MI POLLO INC   
22800   0003012425-0001-6          DD RESTAURANT HOLDINGS INC   
22963   0000840443-0001-9                         SHIN HO KIM   
24303   0002679265-0001-1                          PUN OK LEE   
24687   0003004124-0001-5               CBR ENTERTAINMENT INC   
25266   0000095013-0001-0                      B & C YOON INC   
27812   0002306811-0001-1                      HYUN SOON KANG   
28273   0000486616-0001-5  CHANG'S RESTAURANT ENTERPRISES INC   
30707   0002736611-0001-6                       YOUNG JA PARK   
31825   0000963909-0001-2                     KWANG SOON SHIM   
32595   0000079201-0001-0                      GAMJA BAWI INC   
34109   0000243243-0001-1                MARIO CRUZ HERNANDEZ   
34139   0000101586-0001-9                       BBB GROUP LLC   
...                   ...                                 ...   
495955  0002916081-0001-1                     ONGGANAE LA INC   
496682  0002985248-0001-1                           MOKIA INC   
497044  0002769822-0001-1                          GYB LA LLC   
497141  0003005745-0001-3                   CAROLINA MARTINEZ   
497675  0002959418-0001-7                          SSKH, INC.   
497705  0003001313-0001-1                            HMCU INC   
497812  0002657735-0001-4                 HERA ENTERPRISE INC   
499354  0002822064-0001-9              HONESPLE ENTERTAINMENT   
500135  0003003638-0001-1           HUI YAN INTERNATIONAL LLC   
500527  0002538189-0001-1                           ANDY MOON   
500641  0002994719-0001-5               TWIST BAKE SHOP, INC.   
501351  0002970940-0001-0                       UWSTE 200 INC   
502055  0002372215-0003-2                        NUKI SUB INC   
502166  0002979005-0001-0                PHARAOH KINGDOM, INC   
502369  0002960711-0001-1                          APCKT CORP   
503182  0002745168-0001-4                    MONA TRADING INC   
503810  0002946654-0001-1                1076 OPERATIONS, LLC   
504547  0002569664-0001-8                        A PHO 21 INC   
504894  0002937544-0001-1                          JISUNG CHA   
506659  0002991721-0001-1                           FWU S KUO   
507300  0003087717-0001-7                        SOUFFLES INC   
507389  0003029879-0001-2                        T-FISH, INC.   
507449  0002789624-0001-7                  DIDIM FOOD USA INC   
507699  0000260464-0013-4            CARLS JR RESTAURANTS LLC   
508555  0000653062-0001-6                         JIA ZEN INC   
508644  0002828440-0001-0                           ATJ&B LLC   
509546  0003091048-0001-8                   PHO CAFE PLUS INC   
510181  0003092887-0001-1                       RED PIG, INC.   
510772  0003093214-0001-0             ERWIN OVIDIO REYES RUIZ   
512903  0003091634-0001-5                 

Records in the 'Full-service restaurants', 'Unclassified establishments (unable to classify)', 'All other miscellaneous store retailers (including tobacco, candle, & trophy shops)' and 'Limited-service eating places' categories may also yield answers to the rest of my burning questions.

## How many Korean BBQ restaurants are in Koreatown?

Upon closer investigation, not all Koreatown Korean barbecue restaurants are called BBQ: 
* 'Galbi/Galbee/Kalbi' and 'Bulgogi' are commonly found in the names of Korean BBQ restaurants. _Galbi_ means 'ribs' and _bulgogi_ refers to marinated meats.
* Many such restaurants are named for their specialty, i.e.: Ahgassi Gopchang (_gopchang_ meaning 'small intestines').
* One establishment is called 'Ssam'. _Ssam_ is the word for lettuce or soaked rice paper wrapped around grilled meats.
* At least one restaurant devoted to Korean BBQ is called 'Barbeque'.
* Soot Bull Jeep and The Corner Place are both Korean BBQ restaurants in K-town that have BBQ nowhere in the name.

The NAICS codes will not be as helpful as I thought for restaurants, as many are miscategorized or lack a NAICS code entirely.

In [77]:
ktown_bbq = ktown_biz.loc[(ktown_biz['dba_name'].str.contains('BBQ|BARBEQUE|GALBI|GALBEE|SSAM|BULGOGI|GOPCHANG|GOBCHANG') == True) 
                          | (ktown_biz['business_name'].str.contains('BBQ|BARBEQUE|GALBI|GALBEE|SSAM|BULGOGI|GOPCHANG|GOBCHANG') == True), :]
ktown_bbq

location_account_#                       business_name  \
7406    0002738450-0001-4                SSAM RESTAURANT, INC   
14793   0000558001-0001-1        CHOSUN GALBEE RESTAURANT INC   
17819   0002361518-0001-1               YOUNG JIN CORPORATION   
46665   0003021202-0001-8                         JAMES T KIM   
62074   0003021631-0001-7                          GUHARA INC   
64554   0003025243-0001-6                         ACEPLUS INC   
90053   0002569863-0001-0                 JHY ENTERPRISES INC   
98921   0002721014-0001-8                         KNUSAHA LLC   
105464  0000934936-0002-6                     BUDNAMU USA INC   
113481  0002648250-0001-9                            CYTO INC   
120236  0002321879-0001-6      L SQUARE INVESTMENT GROUP, INC   
144820  0002773482-0001-3                        JONG KUK LEE   
170099  0002630468-0001-1                     HONG GROUPS INC   
186826  0002193188-0001-6                       HOSSAM K TAHA   
189188  0000801414-0001-4            MCNAMARA INVESTMENT, INC   
201927  0002361786-0001-3                        HYUNRAN CHIN   
213569  0002756464-0001-4                      BBQ OOKOOK INC   
229764  0002523280-0001-5                        CHUN OK HONG   
231799  0002532256-0001-1                       SANYA BBQ INC   
237189  0002722657-0001-2                   ANTOINETTE MESSAM   
242762  0002710997-0001-1              HYUNDAE CAR WASH, INC.   
246067  0003044580-0001-7                T.U.J. KUMHINO CORP.   
253323  0003058574-0001-7                       SEOUL BBQ INC   
280205  0002770962-0001-5                       LB GLOBAL INC   
284376  0003064789-0001-7                      KOREAN BBQ INC   
293713  0002674753-0001-3                      ISSAM T DAHDUL   
304860  0003071240-0001-5                       KANS 75, INC.   
319953  0002746496-0001-8                   TOUR ANGELES, INC   
321223  0003074195-0001-0         607 WESTERN RESTAURANTS LLC   
332246  0003034981-0001-8                      DAMEIN JACKSON   
341183  0002853918-0001-9  LA HAWAIIAN BBQ & CHINESE FOOD INC   
341501  0002826921-0001-3                  GALBI BROTHERS INC   
365465  0002569380-0001-2                     JSP VENTURE INC   
392750  0002899454-0003-0                   ISSAM MOUSA RAHIB   
409755  0002931285-0001-1                    BIRK & KACH, LLC   
413311  0002958666-0001-5          NAM SONG DEVELOPMENT, INC.   
416529  0002822915-0001-9               LOS ORACLE MGMT, INC.   
429652  0002815890-0001-2                      HIATUS BBQ INC   
431022  0002788860-0001-2                   ABBA TRADING CORP   
441712  0002995276-0001-0                  URBAN BARBECUE LLC   
467636  0002914625-0001-1                     BREAKERS VC LLC   
467988  0002958119-0001-0                      EUNSANG 75 INC   
471547  0002699613-0001-0                AHGASSI GOPCHANG LLC   
473165  0002983700-0001-7                       DYL BBQ, INC.   
476820  0002937614-0001-8                           MIDRI INC   
478955  0002913616-0001-5                  MAPO CHICKEN, INC.   
481775  0002946132-0001-4                   FERNANDO CARRILLO   
482262  0002639469-0001-1                         MIDRI, INC.   
497044  0002769822-0001-1                          GYB LA LLC   
507449  0002789624-0001-7                  DIDIM FOOD USA INC   

                                   dba_name  \
7406                                    NaN   
14793                                   NaN   
17819                      EIGHT KOREAN BBQ   
46665                STONE GRILL KOREAN BBQ   
62074                         WI KOREAN BBQ   
64554                       MONG KOREAN BBQ   
90053                           STAR BBQ #1   
98921                             COMET BBQ   
105464                   BUD NAMU BBQ HOUSE   
113481                        U2 KOREAN BBQ   
120236                     LA SOO WON GALBI   
144820                        U2 KOREAN BBQ   
170099                           HONG GALBI   
186826                           

In [79]:
ktown_bbq.shape

(50, 12)

This query yields a few false positives, as well as great Korean restaurants that don't serve barbeque, so we will sift those out:

In [71]:
ktown_bbq = ktown_bbq.drop([186826,189188,201927,237189,321223,332246,341183,392750,409755,431022,482262,429652,481775], axis=0)

In [72]:
ktown_bbq.shape

(37, 12)

The sifted list does not include businesses that specialize in KBBQ but lack barbecue or variations thereof in the name, so we will add those. Searching by address proves most fruitful of all, as some restaurants are not listed in this index by the name under which patrons know them. 
* A K-BBQ restaurant called 'Soot Bull Jeep', one of LA's only charcoal grill establishments, appears in the business listings as 'SBJP INC'. 
* Park's BBQ, a restaurant that routinely makes LA's best Korean barbeque lists, is indexed as 'PARK DAE GAM NE INC'.
* Quarters Korean BBQ, a restaurant that shares a plaza with Kang Hodong Baekjeong, another K-BBQ house, is listed as simply 'QUARTERS'. According to [Eater Los Angeles](https://la.eater.com/maps/korean-bbq-map-to-try-right-now-koreatown-los-angeles-map-guide), Quarters is a bit more tourist-friendly and _owned by Kang Hodong_.

In [88]:
ktown_biz.loc[(
    (ktown_biz['dba_name'].str.contains('CORNER PLACE|DAE GAM|BAEK JEONG|DONG IL|BAK KUNG') == True) | 
    (ktown_biz['business_name'].str.contains('SBJP|CH WILSHIRE|CLAYTON AND') == True)), :]

location_account_#                  business_name  \
16830   0003010987-0001-3  CLAYTON AND CASEY CORPORATION   
37364   0000417392-0001-4                     INKEUM KIM   
71795   0000167584-0001-1          DONG IL JANG REST INC   
104639  0002619289-0001-9                        678 INC   
109799  0000019766-0001-7            PARK DAE GAM NE INC   
121585  0002875617-0001-7                       SBJP INC   
213449  0002702442-0001-7  CH WILSHIRE INTERNATIONAL INC   
395088  0002598350-0001-3                    WON HUA INC   
426861  0002598350-0002-1                    WON HUA INC   

                              dba_name              street_address  \
16830                              NaN         2561 W OLYMPIC BLVD   
37364                 THE CORNER PLACE      2819 JAMES M WOOD BLVD   
71795                     DONG IL JANG           3455 W 8TH STREET   
104639                      BAEK JEONG  3465 W 6TH STREET UNIT #20   
109799  CHUNG DAM DONG PARK DAE GAM NE     955 S VERMONT AVENUE #G   
121585                             NaN           3136 W 8TH STREET   
213449                             NaN           3819 W 6TH STREET   
395088                      BAK KUNG 1  3700 W OLYMPIC BLVD   #100   
426861                      BAK KUNG 2        233 S VERMONT AVENUE   

               city zip_code          location_description     naics  \
16830   LOS ANGELES    90006       2561 OLYMPIC 90006-2986  722110.0   
37364   LOS ANGELES    90006  2819 JAMES M WOOD 90006-1701  722110.0   
71795   LOS ANGELES    90005           3455 8TH 90005-2517  722110.0   
104639  LOS ANGELES    90020           3465 6TH 90020-2567  722110.0   
109799  LOS ANGELES    90006        955 VERMONT 90006-1647  722110.0   
121585  LOS ANGELES    90005           3136 8TH 90005-1903  722110.0   
213449  LOS ANGELES    90020           3819 6TH 90020-3901  722110.0   
395088  LOS ANGELES    90019       3700 OLYMPIC 90019-2040  722300.0   
426861  LOS ANGELES    90004        233 VERMONT 90004-5906  722110.0   

                                primary_naics_description  council_district  \
16830                            Full-service restaurants                 1   
37364                            Full-service restaurants                 1   
71795                            Full-service restaurants                10   
104639                           Full-service restaurants                10   
109799                           Full-service restaurants                10   
121585                           Full-service restaurants                10   
213449                           Full-service restaurants                10   
395088  Special food services (including food service ...                10   
426861                           Full-service restaurants                13   

       location_start_date              location  
16830           08/01/2017  (34.0526, -118.2869)  
37364           09/01/1982  (34.0558, -118.2897)  
71795           01/01/1979  (34.0576, -118.3049)  
104639          02/02/2012  (34.0635, -118.2969)  
109799          06/13/2003   (34.054, -118.2916)  
121585          12/24/2015  (34.0576, -118.2961)  
213449          09/01/2013  (34.0635, -118.3058)  
395088          01/16/2012    (34.053, -118.319)  
426861          03/06/2014  (34.0699, -118.2916)

In [89]:
ktown_biz.loc[(ktown_biz['street_address']).str.contains('3136 W 8TH ST|955 S VERMONT|3465 W 6TH|3744 W 6TH|233 S VERMONT|3819 W 6TH|3429 W 8TH|1101 S VERMO|2561 W OLYMPIC'), :]

location_account_#                     business_name  \
16830   0003010987-0001-3     CLAYTON AND CASEY CORPORATION   
22612   0002259769-0001-6                  V INVESTMENT INC   
25043   0000791686-0001-7       VUN OUR UNG/SOPHEAK EAP UNG   
27164   0003013193-0008-2                  LYA NEREIRA ROSA   
34323   0003014095-0001-3                            JIE LI   
39011   0002551245-0001-8                      SUNG YEE LEE   
62074   0003021631-0001-7                        GUHARA INC   
72995   0000044169-0001-4                      GERMAN RODAS   
92298   0002785183-0001-4       THE QUARTER POUND HOUSE INC   
97138   0002690771-0001-5                         YHOON INC   
104639  0002619289-0001-9                           678 INC   
105842  0000188327-0001-7                SANG H/BYONG S RHO   
107576  0002685543-0001-7                 MAE SHIL TANG INC   
109799  0000019766-0001-7               PARK DAE GAM NE INC   
121585  0002875617-0001-7                          SBJP INC   
134991  0000483288-0002-3                    RAJ'S SUBS INC   
135594  0000073208-0001-8                   123 WESTERN LLC   
142541  0002997898-0001-0                ROWHEN CORPORATION   
145272  0003042030-0001-3                   955 VERMONT LLC   
180117  0002549467-0001-2             WOORI SHOPPING LA INC   
181676  0002579681-0001-8                    FLAVID E LOPEZ   
190969  0002864561-0001-6                       SE HOON CHO   
203786  0002723693-0001-6              PARKS GLOBAL NET INC   
213449  0002702442-0001-7     CH WILSHIRE INTERNATIONAL INC   
214515  0002262232-0001-8                       MARTA OCHOA   
219503  0002708813-0001-7  S2 NATURAL & ORGANIC CORPORATION   
223907  0002687906-0001-3                   TWITTERLAND INC   
228978  0003040511-0001-7                    SUNG HYOK CHUN   
230502  0002672380-0001-3                  CEST SI BON, INC   
232180  0003053291-0001-0                            KANBAI   
...                   ...                               ...   
291707  0002757396-0001-3                      DONGZHU PIAO   
296155  0000989890-0003-9                        TIGER BANG   
297164  0002173881-0001-9                       TOEBANG INC   
308081  0002865570-0001-2                   SANG YOUN HWANG   
312624  0002780742-0001-8                    WILL & WIN INC   
312709  0003078655-0001-5        EBEN PERMANENT MAKEUP CORP   
326597  0002796544-0001-1                    YOUNG NAM PARK   
339119  0000484038-0001-8              JANET MYUNG YOON LEE   
347735  0002858287-0001-9                        JI RAN WON   
351117  0002816272-0001-2                       SANGHEE HAN   
351622  0003084930-0001-7                  JOKBAL1030, INC.   
364431  0000037459-0013-4             STARBUCKS CORPORATION   
370052  0000012545-0001-8                         J K GROUP   
371162  0000970397-0001-1           SEEDAY HEARING AID CORP   
387160  0003086874-0001-6                   RTG CHAPMAN LLC   
394494  0002974460-0001-5                 SNOW MONSTER INC.   
401512  0002854724-0001-5                      HYUN WOO INC   
406666  0002828829-0001-6                   LC BROTHERS INC   
423877  0002884356-0001-1                        ALLIST INC   
424499  0002723693-0002-4              PARKS GLOBAL NET INC   
426861  0002598350-0002-1                       WON HUA INC   
430349  0002991315-0001-2                       HAN SOK KIM   
440836  0002948127-0002-6                      QI YING WANG   
442045  0002981778-0001-4                   TECH BEAUTY INC   
455187  0002980748-0001-6                        FENGHUA LU   
464669  0002650802-0003-3                        JACKIE CHO   
476770  0002964980-0001-7                    DEL MAR YC INC   
489165  0002999422-0001-9                          STSK INC   
502325  0002956439-0001-7                      WEN JUAN WEI   
507389  0003029879-0001-2                      T-FISH, INC.   

                                   dba_name  \
16830                                   NaN   
22612 

In [ ]:
ktown_bbq.loc[37364, :] = pd.DataFrame(la_biz.loc[la_biz['dba_name'].isin(['THE CORNER PLACE']), :])